base/src/commonMain/kotlin/monitor/ServiceMonitorState.kt

```diff
@@ -0,0 +1,119 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.base.monitor
+
+import com.splendo.kaluga.base.ServiceMonitor
+import com.splendo.kaluga.base.flow.SpecialFlowValue
+import com.splendo.kaluga.state.ColdStateFlowRepo
+import com.splendo.kaluga.state.State
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import kotlin.coroutines.CoroutineContext
+
+sealed class ServiceMonitorState : State() {
+
+    sealed class Initialized(
+        private val monitor: ServiceMonitor
+    ) : ServiceMonitorState() {
+        val notInitialized: suspend () -> NotInitialized = { NotInitialized(monitor) }
```

Comment:
I have doubts on making methods and fields of the state public. it leaks implementation details in my opinion. could you mark them as internal? or even better - rewrite to make the transitions private

Reply:
987ec4a